client/web/src/repo/tree/TreePagePanels.tsx

```diff
@@ -0,0 +1,322 @@
+import React, { useCallback, useRef, useState } from 'react'
+
+import { mdiFileDocumentOutline, mdiFolderOutline, mdiMenuUp, mdiMenuDown } from '@mdi/js'
+import classNames from 'classnames'
+import * as H from 'history'
+
+import { LinkOrSpan } from '@sourcegraph/shared/src/components/LinkOrSpan'
+import { TreeFields } from '@sourcegraph/shared/src/graphql-operations'
+import { StackedMeter, Link, Icon, Card, CardHeader, ParentSize } from '@sourcegraph/wildcard'
+
+import { RenderedFile } from '../blob/RenderedFile'
+
+import styles from './TreePage.module.scss'
+
+interface ReadmePreviewCardProps {
+    readmeHTML: string
+    readmeURL: string
+    location: H.Location
+}
+
+export const ReadmePreviewCard: React.FunctionComponent<ReadmePreviewCardProps> = ({
+    readmeHTML,
+    readmeURL,
+    location,
+}) => {
+    const fileRef = useRef<HTMLDivElement>(null)
+    const containerRef = useRef<HTMLDivElement>(null)
+    const isNotCutoff =
+        fileRef.current &&
+        containerRef.current &&
+        fileRef.current.clientHeight > 0 &&
+        containerRef.current.clientHeight >= fileRef.current.clientHeight
```

Comment:
This approach seems to cause a "flicker" - for example, when viewing a directory like:

```
repo:^github\.com/sourcegraph/sourcegraph$ file:^cmd
```

The file when render, the "More..." button will render briefly, and then it will disappear. I think it's because this particular file ends up with a height that's almost exactly the height of the container?

Reply:
It looks like it behaves a bit better if we add a bit of a padding to the condition:

```suggestion
    const fileRef = useRef<HTMLDivElement>(null)
    const containerRef = useRef<HTMLDivElement>(null)
    const isNotCutoff =
        fileRef.current &&
        containerRef.current &&
        fileRef.current.clientHeight > 0 &&
        containerRef.current.clientHeight >= fileRef.current.clientHeight - 4
```